### PR TITLE
publish de/serialization

### PR DESCRIPTION
**Pull request by YannickWeber**

### CODE REVIEW

File: src/main/java/org/varroa/load/Mqtt5Slave.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package org.varroa.load;
```
Comment (s-gheldd) [inline src/main/java/org/varroa/load/Mqtt5Slave.java]: Class is not part of the work packet. Please remove from PR

File: src/main/java/org/varroa/load/MqttContext.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package org.varroa.load;
```
Comment (s-gheldd) [inline src/main/java/org/varroa/load/MqttContext.java]: Class is not part of the work packet. Please remove from PR

File: src/main/java/org/varroa/load/MqttSlave.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package org.varroa.load;
```
Comment (s-gheldd) [inline src/main/java/org/varroa/load/MqttSlave.java]: Class is not part of the work packet. Please remove from PR

File: src/main/java/org/varroa/remote/codec/decoder/DecodeUtils.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+import io.netty.buffer.ByteBuf;
+import org.jetbrains.annotations.NotNull;
+
+import java.nio.charset.StandardCharsets;
+
+/**
+ * @author Yannick Weber
+ */
+public class DecodeUtils {
```
Comment (SgtSilvio) [inline src/main/java/org/varroa/remote/codec/decoder/DecodeUtils.java]: 👍

**Comment by SgtSilvio** [comment]
I changed the base branch to the `connect-command` branch as the review is then easier.
This should probably not be merged into the `connect-command` branch, so the base must be changed to `master` back again after the `connect-command` branch is merged.

File: src/main/java/org/varroa/chunk/command/PublishCommand.java [hunk 5]
```diff
@@ -6,27 +6,50 @@
...
+        this.waitForAck = waitForAck;
     }
 
     public @NotNull MqttQos getQos() {
         return qos;
     }
 
-    public boolean isRetain() {
-        return retain;
+    public boolean waitForAck() {
```
Comment (SgtSilvio) [inline src/main/java/org/varroa/chunk/command/PublishCommand.java]: minor: should be called `isWaitForAck` (convention for boolean getters) as `waitForAck` implies that you start the action and wait for the acknowledgement now.

File: src/main/java/org/varroa/remote/codec/encoder/EncodeUtils.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+import io.netty.buffer.ByteBuf;
+import org.jetbrains.annotations.NotNull;
+
+import java.nio.charset.StandardCharsets;
+
+/**
+ * @author Yannick Weber
+ */
+public class EncodeUtils {
```
Comment (SgtSilvio) [inline src/main/java/org/varroa/remote/codec/encoder/EncodeUtils.java]: 👍

File: src/main/java/org/varroa/remote/codec/encoder/command/CommandEncoders.java [hunk 7]
```diff
@@ -6,18 +6,29 @@
...
 
+/**
+ * @author Yannick Weber
+ * @author Georg Held
+ */
 @Singleton
 public class CommandEncoders {
 
     private final @NotNull Provider<ConnectEncoder> connectEncoderProvider;
+    private Provider<PublishEncoder> publishEncoderProvider;
```
Comment (SgtSilvio) [inline src/main/java/org/varroa/remote/codec/encoder/command/CommandEncoders.java]: `final @NotNull` missing

**Pull request merged**